content/blog/embrace-kubernetes-part1/index.md

```diff
@@ -0,0 +1,96 @@
+
+---
+
+When you’re considering whether or not to implement Kubernetes, perhaps the first question to ask yourself is do you need it at all?
+
+The point of any technology isn’t the technology itself. The point is whether you can use it to reduce the barrier of entry for application developers so they can get features from their machines to your customers as quickly and easily as possible. Do you already have a solution that works well? If you do, why do you want to change it? Making such a radical change in your technology is potentially quite dangerous so what’s your motivation?
+
+It might be wiser to stay with the solution you have and improve it. Also, don't fall into the trap of believing that any technology, not just Kubernetes, will be a magical answer to all your problems, whether organizational or technical. There’s no such thing.
+
+In this blog post we’ll talk about how to evaluate your own situation to see if Kubernetes is a good fit. We’ll see that the question isn’t that simple to answer and there are a lot of variables to consider. In the next blog post, we’ll talk about a situation where Kubernetes can be a good fit and how to start your first Kubernetes project.
+
+Note that these blog posts assume you already have some familiarity with Kubernetes. If you need some introductory material, the [Kubernetes web site](https://kubernetes.io/docs/concepts/overview/what-is-kubernetes/) is a good place to start.
+
+<!--more-->
+
+## Why is Kubernetes So Popular?
+
+Kubernetes, an open-source system for deploying, scaling, and managing containerized applications, is currently the last step in an evolutionary chain that began with private data centers and, from there, went to cloud VMs, then to containers and then to container orchestrators. Kubernetes was developed by Google. Version 1.0 was released in 2015 and the latest version was released in October 2020.
+
+Kubernetes has become increasingly possible over the last few years. In the ZDNet article, [“Kubernetes jumps in popularity”](https://www.zdnet.com/article/kubernetes-jumps-in-popularity/), published on March 6, 2020, Steven J. Vaughan-Nichols cites a [Cloud Native Computing Foundation](https://www.cncf.io/) (CNCF) survey about containers and Kubernetes. Of the companies that responded, 84% were using containers and, of those companies, 78% were using Kubernetes to manage them. The survey also found that there were over 109 tools to manage containers and 89% of them use various versions of Kubernetes.
+
+There are several reasons why Kubernetes is so widely adopted. Here are some of them:
+
+- **Infrastructure as data**. Many configuration management tools treat infrastructure as code. This generally means learning a domain-specific language or a particular programming language. Kubernetes treats infrastructure as data, which means it can be expressed as YAML files. There are pros and cons to using configuration files so you and your team will have to discuss this.
```

Comment:
I personally wouldn't emphasize this. It seems counter to our own brand, no? And arguably, this isn't the most important point, instead it's that Kubernetes is built around the entire notion of declarative goal state and eventual consistency. That makes it amenable to declarative IaC techniques via YAML, but the underpinning reason is more fundamental, and is precisely why Pulumi was able to have amazing Kubernetes support -- the underlying models are an exact match for one another. I'd rather tee this up.....

Reply:
removed